internal/database/metadata/store.go

```diff
@@ -21,7 +21,7 @@ type ReadStore interface {
 	CacheGetEntity(ctx context.Context, id int) (*types.Entity, error)
 	CacheGetEntityByName(ctx context.Context, name string) (*types.Entity, error)
 	CacheListEntity(ctx context.Context) types.EntityList
-	ListEntity(ctx context.Context) (types.EntityList, error)
+	ListEntity(ctx context.Context, entityIDs *[]int) (types.EntityList, error)
```

Comment:
Maybe `ListEntity(ctx context.Context, entityIDs []int) (types.EntityList, error)` is better and there can be less checks in implement 🤔

Reply:
`entityIDs == nil` and `len(*entityIDs) == 0` represents two different conditions:
- `entityIDs == nil` means list all entities
- `len(*entityIDs) == 0`  means don't no entity

that's why we need a pointer here, see `ListFeatureOpt` for reference

Reply:
well, understand. we need to know the user if used argument `entityIDs`

Reply:
yup, that's correct!

Reply:
I feel that there are two things that are not so good about this way of writing:
1. pointer slices are not well constructed.
2. the method definition does not make it clear to the user how the `entityIDs` affect the result.

How do you feel about this?
```go
type ListEntityOpt struct {
      EntityIDs []int
      UseEntityIds bool
}
```

Reply:
hmm introducing another argument doesn't alleviate the complexity here. If we have more than one condition, we will double the arguments in `ListEntityOpt`

Reply:
make sense